packages/facility-server/app/sync/FacilitySyncManager.js

```diff
@@ -156,11 +157,21 @@ export class FacilitySyncManager {
       startedAtTick: newSyncClockTime,
     });
 
-    await this.pushChanges(sessionId, newSyncClockTime);
-
-    await this.pullChanges(sessionId);
-
-    await this.centralServer.endSyncSession(sessionId);
+    try {
+      await this.pushChanges(sessionId, newSyncClockTime);
+
+      await this.pullChanges(sessionId);
+      await this.centralServer.endSyncSession(sessionId);
+    } catch (error) {
+      if (!(error instanceof Problem)) {
+        // if the error is not a Problem, it occurred locally on the facility-server and we should notify the central server
+        await this.centralServer.markSessionErrored(sessionId, error.message);
+      }
+      throw error;
```

Comment:
@passcod could I get your input on this? My impression with the `Problem` class is that it's a wrapper around errors from the TamanuApi. So if an error is a `Problem`, we can assume it came from the TamanuApi?

Probably ideally we'd implement a specific `Problem` type for sync errors that came from central-server and check for that, but I wasn't feeling like making all those changes right now.